File: sdk-dotnet/OtpManagerException.cs
using System;

namespace OtpManager.Client
{
    /// <summary>
    /// Strongly typed exception thrown by the OTP Manager SDK.
    /// </summary>
    public class OtpManagerException : Exception
    {
        /// <summary>
        /// Gets the OTP Manager error code (e.g. "INVALID_TOKEN", "ACCOUNT_LOCKED").
        /// </summary>
        public string Code { get; }

        /// <summary>
        /// Gets the HTTP Status Code associated with the error.
        /// </summary>
        public int StatusCode { get; }

        /// <summary>
        /// Initializes a new instance of the <see cref="OtpManagerException"/> class.
        /// </summary>
        public OtpManagerException(string code, string message, int statusCode = 400)
            : base($"[{code}] {message} (HTTP {statusCode})")
        {
            Code = code;
            StatusCode = statusCode;
        }
    }
}

File: sdk-java/src/main/java/com/otpmanager/client/exception/OtpManagerException.java
package com.otpmanager.client.exception;

/**
 * Strongly typed exception thrown by the OTP Manager Java SDK.
 */
public class OtpManagerException extends RuntimeException {

    private final String code;
    private final int statusCode;

    public OtpManagerException(String code, String message, int statusCode) {
        super("[" + code + "] " + message + " (HTTP " + statusCode + ")");
        this.code = code;
        this.statusCode = statusCode;
    }

    public String getCode() {
        return code;
    }

    public int getStatusCode() {
        return statusCode;
    }
}
